diff --git a/dist/plugins/one-time-token/index.d.ts b/dist/plugins/one-time-token/index.d.ts
index 4f7655ba2b4e2dec9d8152642befa8c7eee219ec..86eb7922fd2e022548d9f17ecd2765cdeb61e74c 100644
--- a/dist/plugins/one-time-token/index.d.ts
+++ b/dist/plugins/one-time-token/index.d.ts
@@ -7,7 +7,7 @@ import 'jose';
 import 'kysely';
 import 'better-sqlite3';
 
-interface OneTimeTokenOptions {
+export interface OneTimeTokenOptions {
     /**
      * Expires in minutes
      *
