diff --git a/ios/Sources/ObjcHelpers/RNIObjcUtils.mm b/ios/Sources/ObjcHelpers/RNIObjcUtils.mm
index 7931a0e69882683e01ca2e486d75a47afc506970..38f71a7f00d651e57ba203050b9042b84f1c85ab 100644
--- a/ios/Sources/ObjcHelpers/RNIObjcUtils.mm
+++ b/ios/Sources/ObjcHelpers/RNIObjcUtils.mm
@@ -19,7 +19,7 @@
 #if __cplusplus
 #import <folly/dynamic.h>
 #import <React/RCTConversions.h>
-#import <React/RCTFollyConvert.h>
+#import "RNIUtilitiesFollyConvert.h"

 #include <react/renderer/core/LayoutMetrics.h>
 #include <react/renderer/graphics/Rect.h>
diff --git a/ios/Sources/RNIBaseView/RNIBaseView.mm b/ios/Sources/RNIBaseView/RNIBaseView.mm
index 3ab197a524959ffd8cab8ed55d0df952f8f9000c..7aeb0eb72ad15db55bf7d1fff7a0d0238c295fe2 100644
--- a/ios/Sources/RNIBaseView/RNIBaseView.mm
+++ b/ios/Sources/RNIBaseView/RNIBaseView.mm
@@ -27,7 +27,7 @@

 #import "RCTFabricComponentsPlugins.h"

-#import <React/RCTFollyConvert.h>
+#import "RNIUtilitiesFollyConvert.h"
 #import <React/RCTSurfaceTouchHandler.h>

 #include <react/renderer/core/ConcreteComponentDescriptor.h>
diff --git a/ios/Sources/RNIUtilitiesModule/RNIUtilitiesFollyConvert.h b/ios/Sources/RNIUtilitiesModule/RNIUtilitiesFollyConvert.h
new file mode 100644
index 0000000000000000000000000000000000000000..bdf7ca90c1f224f0704acf65353cf881751724f6
--- /dev/null
+++ b/ios/Sources/RNIUtilitiesModule/RNIUtilitiesFollyConvert.h
@@ -0,0 +1,15 @@
+#ifdef RCT_NEW_ARCH_ENABLED
+
+#if __has_include(<react/utils/FollyConvert.h>)
+  // static libs / header maps (no use_frameworks!)
+  #import <react/utils/FollyConvert.h>
+#elif __has_include("FollyConvert.h")
+  // `use_frameworks! :linkage => :static` users will need to import FollyConvert this way
+  #import "FollyConvert.h"
+#elif __has_include("RCTFollyConvert.h")
+  #import "RCTFollyConvert.h"
+#else
+  #error "FollyConvert.h not found. Ensure React-utils & RCT-Folly pods are installed."
+#endif
+
+#endif
diff --git a/ios/Sources/RNIUtilitiesModule/RNIUtilitiesModule.mm b/ios/Sources/RNIUtilitiesModule/RNIUtilitiesModule.mm
index 097658fe17334118ca54e9e6e5225ac2ed15d434..0e53f8489c5d1fbeba5325071da985b16490507b 100644
--- a/ios/Sources/RNIUtilitiesModule/RNIUtilitiesModule.mm
+++ b/ios/Sources/RNIUtilitiesModule/RNIUtilitiesModule.mm
@@ -19,7 +19,7 @@

 #import <folly/dynamic.h>
 #import <React/RCTConversions.h>
-#import <React/RCTFollyConvert.h>
+#import "RNIUtilitiesFollyConvert.h"

 #include <jsi/jsi.h>
 #include <string>
